common4j/src/main/com/microsoft/identity/common/java/interfaces/IStorageSupplier.kt

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+package com.microsoft.identity.common.java.interfaces
+
+import com.microsoft.identity.common.java.cache.IMultiTypeNameValueStorage
+
+/**
+ * An interface for loading [INameValueStorage]
+ */
+interface IStorageSupplier {
```

Comment:
Can we simplify this by only exposing one method to get NameValueStore
I don't think we are using the specific typed or multitype NameValue storage.

Reply:
yes... make sense.

Reply:
We're using Long storage in linux.

Reply:
What is the difference between. NameValue and File Store?
When do we use one vs the other.

Reply:
One is specific to a certain type, the other is more flexible.

Technically we could convert all usage of nameValue to the latter, let's see how much of the change that is....

Reply:
So, in Linux we are using both one generic one that can support more than 1 type and the other is specific typed. (Long) ?

Reply:
@iamgusain yes